packages/react/src/hooks/program.ts

```diff
@@ -2,53 +2,179 @@
 
 "use client";
 
-import { useMutation, type UseMutationOptions, useQueryClient } from "@tanstack/react-query";
+import {
+  useMutation,
+  type UseMutationOptions,
+  useQuery,
+  useQueryClient,
+  type UseQueryOptions,
+} from "@tanstack/react-query";
 import type { Address, Instruction } from "gill";
+import {
+  type Signature,
+  type SolanaClient,
+  type TransactionSendingSigner
+} from "gill";
 
 import { GILL_HOOK_CLIENT_KEY } from "../const.js";
 
-type InstructionBuilder<TInput = any> = (input: TInput, config?: { programAddress?: Address }) => Instruction;
+type InstructionBuilder<TInput = any> = (input: TInput, config?: { programAddress: Address }) => Instruction;
 
-type ProgramHookConfig<TInstructions extends Record<string, InstructionBuilder<any>>> = {
+type AccountFetcher<TData = any> = (rpc: SolanaClient["rpc"], address: Address, config?: any) => Promise<TData>;
+
+type CommitmentLevel = "confirmed" | "finalized";
+
+type SignAndSendFn = (
+  instruction: Instruction,
+  signer: TransactionSendingSigner,
+) => Promise<string>;
+
+type ProgramHookConfig<
+  TInstructions extends Record<string, InstructionBuilder<any>>,
+  TAccounts extends Record<string, AccountFetcher<any>>,
+> = {
+  accounts: TAccounts;
+  commitment?: CommitmentLevel | null;
   instructions: TInstructions;
-  programAddress?: Address;
+  programAddress: Address;
```

Comment:
The programAddress is only used for the queryKey since I removed it from the instruction. So it might not strictly be necessary to include but it works well for the queryKey.